src/main/java/org/junit/runners/ParentRunner.java

```diff
@@ -363,13 +364,60 @@ public void run(final RunNotifier notifier) {
             statement.evaluate();
         } catch (AssumptionViolatedException e) {
             testNotifier.addFailedAssumption(e);
-        } catch (StoppedByUserException e) {
-            throw e;
         } catch (Throwable e) {
-            testNotifier.addFailure(e);
+            processExecutionFailures(testNotifier, gatherAllFailures(e));
         }
     }
 
+    /**
+     * Goes through all execution failures trying to find and re-throw 
+     * {@link StoppedByUserException} while adding all other errors to
+     * test notifier.
+     * 
+     * @param testNotifier notifier to add failures
+     * @param failures list of execution failures
+     */
+    private void processExecutionFailures(EachTestNotifier testNotifier,
+            List<Throwable> failures) {
+        StoppedByUserException stoppedByUserException = null;
+        
+        for (Throwable failure : failures) {
+            if (failure instanceof StoppedByUserException) {
+                stoppedByUserException = (StoppedByUserException) failure;
+            } else {
+                testNotifier.addFailure(failure);
```

Comment:
> This is a behavior change. You are sending to the notifier each of the failures ...

Behavior change from what point of view? Before and after my changes no clients are getting multiple exceptions at once. As I said in item 2 in my pull request message, class `EachTestNotifier` does the same as I do - it unpacks all exceptions and notifies clients about them one by one.

> It sounds the problem is that a StoppedByUserException wrapped ...

Sending `MultipleFailureException` to notifier before finding `StoppedByUserException` will lead to behavior changes. Clients will get notifications about `StoppedByUserException` what is never happened before. My first item in the pull request message was about that. If we are ok with these behavior changes, then it can simplify the code.

> Also, where is the MultiFailureException being created that contains ...

You can get full example from #1125 or check new test I've added. Basically, `StoppedByUserException` is combined with other exceptions while evaluating 'after' methods. I don't think that this is an error, since JUnit guarantees to invoke all 'after' methods in case of user stop request and if any of 'after' methods will throw exception I believe clients should get notification about them.